resources/measures/upgrade_hvac_condensing_boiler/measure.rb

```diff
@@ -0,0 +1,251 @@
+
+# start the measure
+class CondensingBoilers < OpenStudio::Measure::ModelMeasure
+  require 'openstudio-standards'
+
+  # human readable name
+  def name
+    # Measure name should be the title case of the class name.
+    'condensing_boilers'
+  end
+
+  # human readable description
+  def description
+    'This measure replaces an exising natural gas boiler with a condensing gas boiler.'
+  end
+
+  # human readable description of modeling approach
+  def modeler_description
+    'This measure replaces an exising natural gas boiler with a condensing gas boiler. The measure loops through existing boiler objects and increases the efficiency, lowers the water supply temperature, and modifies the performance curve to represent condensing boilers.'
+  end
+
+  ## USER ARGS ---------------------------------------------------------------------------------------------------------
+  # define the arguments that the user will input
+  def arguments(_model)
+    OpenStudio::Measure::OSArgumentVector.new
+  end
+  ## END USER ARGS -----------------------------------------------------------------------------------------------------
+
+  # define what happens when the measure is run
+  def run(model, runner, user_arguments)
+    super(model, runner, user_arguments)
+
+    # report initial condition of model
+    num_boilers = model.getBoilerHotWaters.size
+    runner.registerInitialCondition("The building started with #{num_boilers} hot water boilers.")
+
+    # check for existence of water heater boiler
+    if model.getBoilerHotWaters.empty?
+      runner.registerAsNotApplicable('No hot water boilers found in the model. Measure not applicable ')
+      return true
+    end
+
+    # set new boiler supply temp to 140F to represent condensing boiler
+    hw_setpoint_f = 180
+    hw_setpoint_c = OpenStudio.convert(hw_setpoint_f, 'F', 'C').get
+
+    # set delta T for hw loop to 22.2 C, which corresponds to 100 F return temp
+    hw_delta_t_c = 22.2
+    hw_return_temp_c = hw_setpoint_c - hw_delta_t_c
+
+    # create new setpoint manager for new supply temp of 140F
+    new_setpoint_sched = OpenStudio::Model::ScheduleConstant.new(model)
+    new_setpoint_sched.setValue(hw_setpoint_c)
+    new_setpoint_sched.setName('Condensing Boiler Heating Temperature Setpoint')
+
+    # create biquadratic curve and set upper and lower limit
+    condensing_boiler_curve = OpenStudio::Model::CurveBiquadratic.new(model)
+    condensing_boiler_curve.setName('Condensing Boiler Biquadratic Curve')
+    condensing_boiler_curve.setCoefficient1Constant(1.144514)
+    condensing_boiler_curve.setCoefficient2x(-0.02399)
+    condensing_boiler_curve.setCoefficient3xPOW2(-0.01156)
+    condensing_boiler_curve.setCoefficient4y(-0.00439)
+    condensing_boiler_curve.setCoefficient5yPOW2(0.000019)
+    condensing_boiler_curve.setCoefficient6xTIMESY(0.000393)
+    condensing_boiler_curve.setMinimumValueofy(OpenStudio.convert(90, 'F', 'C').get)
```

Comment:
Source for the coefficients please?

Reply:
Derived by @mpraprost from https://www.lochinvar.com/lit/CREST_w_Hellcat_EfficiencyCurve.pdf